src/Microsoft.SqlTools.ServiceLayer/TableDesigner/TableDesignerValidator.cs

```diff
@@ -0,0 +1,263 @@
+                            });
+                        }
+                        else
+                        {
+                            existingColumns.Add(column);
+                        }
+                    }
+
+                    List<string> existingForeignColumns = new List<string>();
+                    for (int j = 0; j < foreignKey.ForeignColumns.Count; j++)
+                    {
+                        var foreignColumn = foreignKey.ForeignColumns[j];
+                        if (existingForeignColumns.Contains(foreignColumn))
+                        {
+                            errors.Add(new ValidationError()
+                            {
+                                Message = string.Format("Foreign column with name '{0}' has already been added to the foreign key '{1}'. Row number: {2}.", foreignColumn, foreignKey.Name, j + 1),
+                                PropertyPath = new object[] { TablePropertyNames.ForeignKeys, i, ForeignKeyPropertyNames.ColumnMapping, j, ForeignKeyColumnMappingPropertyNames.ForeignColumn }
+                            });
+                        }
+                        else
+                        {
+                            existingForeignColumns.Add(foreignColumn);
+                        }
+                    }
+                }
+            }
+            return errors;
+        }
+    }
+
+    public class NoDuplicateConstraintNameRule : ITableDesignerValidationRule
+    {
+        public List<ValidationError> Run(TableViewModel table)
+        {
+            var errors = new List<ValidationError>();
+            var existingNames = new List<string>();
+            for (int i = 0; i < table.ForeignKeys.Items.Count; i++)
+            {
+                var foreignKey = table.ForeignKeys.Items[i];
+                if (existingNames.Contains(foreignKey.Name))
+                {
+                    errors.Add(new ValidationError()
+                    {
+                        Message = string.Format("The name '{0}' is already used by another constraint. Row number: {1}.", foreignKey.Name, i + 1),
+                        PropertyPath = new object[] { TablePropertyNames.ForeignKeys, i, ForeignKeyPropertyNames.Name }
+                    });
+                }
+                else
+                {
+                    existingNames.Add(foreignKey.Name);
+                }
+            }
+
+            for (int i = 0; i < table.CheckConstraints.Items.Count; i++)
+            {
+                var checkConstraint = table.CheckConstraints.Items[i];
+                if (existingNames.Contains(checkConstraint.Name))
+                {
+                    errors.Add(new ValidationError()
+                    {
+                        Message = string.Format("The name '{0}' is already used by another constraint. Row number: {1}.", checkConstraint.Name, i + 1),
```

Comment:
nit, it might be better to clearly state that "the name is already used by another **Check Constraint**"

Reply:
same as above.